generate_from_csv.py

```diff
@@ -163,7 +171,18 @@ def generate_from_csv(args):
                 warnings.warn("{}: {} travelling in time (departing {})".format(
                         arrival, v_name, departure))
 
-            events["vehicle_events"].append({
+            # adjust SoC if delta_soc < min_soc
+            if args.min_soc < delta_soc:
+                trips_above_min_soc += 1
+                if last_arrival_event is None:
+                    # initially unconnected: adjust initial SoC
+                    vehicles[v_name]["soc"] = delta_soc
+                else:
+                    # adjust last event reference
+                    last_arrival_event["update"]["desired_soc"] = delta_soc
```

Comment:
Probably here also the delta_soc at the moment of a CS available needs to be used.